tests/src/test/scala/system/basic/WskConductorTests.scala

```diff
@@ -0,0 +1,311 @@
+
+  implicit val wskprops = WskProps()
+  val wsk: BaseWsk
+  val allowedActionDuration = 120 seconds
+
+  val testString = "this is a test"
+  val invalid = "invalid#Action"
+  val missing = "missingAction"
+
+  val whiskConfig = new WhiskConfig(Map(WhiskConfig.actionSequenceMaxLimit -> null))
+  assert(whiskConfig.isValid)
+  val limit = whiskConfig.actionSequenceLimit.toInt
+
+  behavior of "Whisk conductor controller"
+
+  it should "invoke a conductor action with no dynamic steps" in withAssetCleaner(wskprops) { (wp, assetHelper) =>
+    val echo = "echo" // echo conductor action
+    assetHelper.withCleaner(wsk.action, echo) { (action, _) =>
+      action.create(
+        echo,
+        Some(TestUtils.getTestActionFilename("echo.js")),
+        annotations = Map("conductor" -> true.toJson))
+    }
+
+    // a normal result
+    val run = wsk.action.invoke(echo, Map("payload" -> testString.toJson))
+    withActivation(wsk.activation, run) { activation =>
+      activation.response.status shouldBe "success"
+      activation.response.result shouldBe Some(JsObject("payload" -> testString.toJson))
+      checkConductorLogsAndAnnotations(activation, 1) // echo
+    }
+
+    // an error result
+    val secondrun = wsk.action.invoke(echo, Map("error" -> testString.toJson))
+    withActivation(wsk.activation, secondrun) { activation =>
+      activation.response.status shouldBe "application error"
+      activation.response.result shouldBe Some(JsObject("error" -> testString.toJson))
+      checkConductorLogsAndAnnotations(activation, 1) // echo
+    }
+
+    // a wrapped result { params: result } is unwrapped by the controller
+    val thirdrun = wsk.action.invoke(echo, Map("params" -> JsObject("payload" -> testString.toJson)))
+    withActivation(wsk.activation, thirdrun) { activation =>
+      activation.response.status shouldBe "success"
+      activation.response.result shouldBe Some(JsObject("payload" -> testString.toJson))
+      checkConductorLogsAndAnnotations(activation, 1) // echo
+    }
+
+    // an invalid action name
+    val invalidrun =
+      wsk.action.invoke(echo, Map("payload" -> testString.toJson, "action" -> invalid.toJson))
+    withActivation(wsk.activation, invalidrun) { activation =>
+      activation.response.status shouldBe "application error"
+      activation.response.result.get.fields.get("error") shouldBe Some(
+        JsString(compositionComponentInvalid(JsString(invalid))))
+      checkConductorLogsAndAnnotations(activation, 1) // echo
+    }
+
+    // an undefined action
+    val undefinedrun = wsk.action.invoke(echo, Map("payload" -> testString.toJson, "action" -> missing.toJson))
+    withActivation(wsk.activation, undefinedrun) { activation =>
+      activation.response.status shouldBe "application error"
+      activation.response.result.get.fields.get("error") shouldBe Some(JsString(compositionComponentNotFound(missing)))
+      checkConductorLogsAndAnnotations(activation, 1) // echo
```

Comment:
is there a separate  integration test for the not entitled case?
how about invoking an action which is actually not an action (eg say a trigger).

Reply:
I have a unit test for this scenario but no integration test. In fact I am tempted to remove the integration tests for error conditions that replicate unit tests because they take a long time. The unit tests only take a few seconds. The integration tests take more than a minute. Thoughts?